axelrod/classifier.py

```diff
@@ -0,0 +1,236 @@
+                cls.all_player_dicts = yaml.load(f, Loader=yaml.FullLoader)
+
+        return cls._instance
+
+    @classmethod
+    def known_classifier(cls, classifier_name: Text) -> bool:
+        """Returns True if the passed classifier_name is known."""
+        global all_classifiers
+        return classifier_name in (c.name for c in all_classifiers)
+
+    @classmethod
+    def __getitem__(
+        cls, key: Union[Classifier, Text]
+    ) -> Callable[[Union[Player, Type[Player]]], Any]:
+        """Looks up the classifier for the player.
+
+        Given a passed classifier key, return a function that:
+
+        Takes a player.  If the classifier is found in the 'classifier' dict on
+        the player, then return that.  Otherwise look for the classifier for the
+        player in the all_player_dicts.  Returns None if the classifier is not
+        found in either of those.
+
+        The returned function expects Player instances, but if a Player class is
+        passed, then it will create an instance by calling an argument-less
+        initializer.  If no such initializer exists on the class, then an error
+        will result.
+
+        Parameters
+        ----------
+        key: A classifier or classifier name that we want to calculate for the
+            player.
+
+        Returns
+        -------
+        A function that will map Player (or Player instances) to their value for
+            this classification.
+        """
+        # Key may be the name or an instance.  Convert to name.
+        if not isinstance(key, str):
+            key = key.name
+
+        if not cls.known_classifier(key):
+            raise KeyError("Unknown classifier")
+
+        def classify_player_for_this_classifier(
+            player: Union[Player, Type[Player]]) -> Any:
+            # If the passed player is not an instance, then try to initialize an
+            # instance without arguments.
+            if not isinstance(player, Player):
+                try:
+                    player = player()
+                except:
+                    pass
+
+            # Factory-generated players won't exist in the table.  As well, some
+            # players, like Random, may change classifiers at construction time;
+            # this get() function takes a player instance, while the saved-values
+            # are from operations on the player object itself.
+            if key in player.classifier:
+                return player.classifier[key]
+
+            try:
+                player_classifiers = cls.all_player_dicts[player.name]
+            except:
+                return None
+
+            return player_classifiers.get(key, None)
+
+        return classify_player_for_this_classifier
+
+    @classmethod
+    def is_basic(cls, s: Union[Player, Type[Player]]):
+        """
+        Defines criteria for a strategy to be considered 'basic'
+        """
+        stochastic = cls.__getitem__("stochastic")(s)
```

Comment:
Does `cls["stochastic"](s)` work here?

Reply:
No.  That was my first try too.

TypeError: 'type' object is not subscriptable